api/types.proto

```diff
@@ -447,13 +447,25 @@ message AcceptancePolicy {
 }
 
 
+message ExternalCA {
+	enum CAType {
+		CFSSL = 0 [(gogoproto.enumvalue_customname) = "CATypeCFSSL"];
+	}
+
+	// CAType is the type of this external CA.
+	CAType type = 1;
+
+	// URL is the URL where the external CA can be reached.
+	string url = 2 [(gogoproto.customname) = "URL"];
```

Comment:
And BTW, I agree that the main urgency is changing it from a list of URLs to a list of structs, so if this is split into separate PRs that seems logical, and please don't take my enthusiasm for the change as pressure in that direction! My main concern from the original PR was simply having the API be reasonably able to support more external CAs than CFSSL before the API became public and hard to change, and I think this is definitely the right direction. The work is much appreciated!